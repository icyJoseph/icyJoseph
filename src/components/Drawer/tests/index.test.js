import React from "react";
import { shallow } from "enzyme";
import { Drawer } from "../";

describe("Drawer", () => {
  const props = {
    visibility: true,
    Content: jest.fn(),
    close: jest.fn()
  };
  const wrapper = shallow(<Drawer {...props} />);
  it("renders", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
