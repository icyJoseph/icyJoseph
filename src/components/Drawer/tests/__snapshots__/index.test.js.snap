// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    Content={[Function]}
    close={[Function]}
    visibility={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "animation": "overlay",
      "as": [Function],
      "children": Array [
        <Segment
          basic={true}
>
          <mockConstructor />
</Segment>,
        <Button
          as="button"
          onClick={[Function]}
>
          Close
</Button>,
      ],
      "direction": "right",
      "style": Object {
        "height": "calc(100vh - 100px) !important",
      },
      "visible": true,
      "width": "wide",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "basic": true,
          "children": <mockConstructor />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "as": "button",
          "children": "Close",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Close",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animation": "overlay",
        "as": [Function],
        "children": Array [
          <Segment
            basic={true}
>
            <mockConstructor />
</Segment>,
          <Button
            as="button"
            onClick={[Function]}
>
            Close
</Button>,
        ],
        "direction": "right",
        "style": Object {
          "height": "calc(100vh - 100px) !important",
        },
        "visible": true,
        "width": "wide",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "basic": true,
            "children": <mockConstructor />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "button",
            "children": "Close",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Close",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
